### ISSUE TITLE
Control inside Viewport inside ViewportContainer ignores gui-event tree order and is in front

### ISSUE THREAD

**Issue by Reneator** [godotengine/godot#48401]
**Godot version:**
v3.3 stable

**OS/device including version:**
Windows 10

**Issue description:**

with a scene-structure like this:
![image](https://user-images.githubusercontent.com/24807557/116878479-b7a5eb80-ac1f-11eb-985c-618551355f23.png)

The TextureRect is drawn in the right order (behind the button), but the TextureRect (set to mouse_filter_stop) catches all the mouse-inputs.
Therefore i cant click on the Button where the TextureRect is being displayed in the ViewportContainer:
![Viewport_Cointainer_Control](https://user-images.githubusercontent.com/24807557/116878445-a9f06600-ac1f-11eb-9846-6faf9a1f3fcd.gif)

In my game i have a world-map which is inside of a viewportContainer where the buttons block mouse-inputs even though they are not visible (either behind another UI-Element or even outside the viewport).
I have a complicated workaround for this in my project, but it still doesnt solve the basic problem of the button being in front.
The workaround checks if the mouse exits the viewport-container (as this control element is still properly handling the mouse-input) and switch the mouse_filter to Ignore for the buttons. But this doesnt work if the mouse is still inside of a button as it leaves the map:
![Viewport_Cointainer_Control_Game](https://user-images.githubusercontent.com/24807557/116879346-da84cf80-ac20-11eb-8f76-3faeb06cdaed.gif)

I was expecting the texture-rect to have the same gui_input order as it is drawn (texture_rect is behind the button)

**Steps to reproduce:**
In the project the native behaviour is replicated. Run the project and see how the textureRect inside the viewport is actually soaks up the mouse-input as if it were in front.

**Minimal reproduction project:**
[Viewportcontainer replication project.zip](https://github.com/godotengine/godot/files/6414931/Viewportcontainer.replication.project.zip)

**Comment by Calinou** [comment]
Related to https://github.com/godotengine/godot/issues/48368.

**Comment by dkaste** [comment]
I think I figured out why this is happening, but I'm not sure if I will have time to fix it.

See the [following lines](https://github.com/godotengine/godot/blob/4e52b84011b4b3c9edb7646731c3ffe451d26316/scene/main/viewport.cpp#L3066) in `Viewport::input()` (linking Godot 4's code, but the same thing applies in 3):
```C++
if (!is_input_handled()) {
    get_tree()->_call_input_pause(input_group, "_input", ev, this); //not a bug, must happen before GUI, order is _input -> gui input -> _unhandled input
}


if (!is_input_handled()) {
    _gui_input_event(ev);
}
```

The code in the first block leads to `SubViewportContainer::_input()` being called, which in turn calls the child's own `Viewport::input()`. The child viewport then runs`_gui_input_event()`, which consumes the event. When the parent viewport exits `_call_input_pause()`, the event as already been marked as handled, so the GUI ignores it.

**Comment by akien-mga** [comment]
Duplicate of #39666
